src/rgw/rgw_rest_s3.cc

```diff
@@ -1713,8 +1713,26 @@ int RGWPostObj_ObjStore_S3::get_policy()
 
     op_ret = rgw_get_user_info_by_access_key(store, s3_access_key, user_info);
     if (op_ret < 0) {
+      class S3Extractor:public RGWS3V2AuthEngine::Extractor {
```

Comment:
It is kind of a makeshift solution, I will have to rework it to extract the tokens also in this class. I plan to do that later. Right now I want to concentrate on making authorize_v2 perfect, then will come back to this